files/edge-core/scripts/launch-edge-core.sh

```diff
@@ -23,6 +23,12 @@ if [ ! -d ${SNAP_DATA}/userdata/mbed ]; then
     mkdir -p ${SNAP_DATA}/userdata/mbed
 fi
 
+# on the remote possibility that we lost power during a factory reset before we
+# could restore mbed credentials, do that now
+if [ -e ${SNAP_COMMON}/mbed-backup ]; then
```

Comment:
I'm assuming that edge-core does not restore this folder correct? They leave that up to the implementer hence why we need to do this?

Reply:
Right, edge-core doesn't know about this folder.  It is created (and restored) in the factory-reset script in an attempt to handle a possible power outage in between the time that the factory-reset script copies the folder and restores the folder while SNAP_DATA is deleted.  That's also why i chose not to copy it to /tmp.